### PR TITLE
Provide access to new settings for HyphenationCompoundWordTokenFilter

### PR DESCRIPTION
**Pull request by pstrsr**
Solves https://github.com/elastic/elasticsearch/issues/97849. 

Adds access to the two new flags `no_sub_matches` and `no_overlapping_matches` to the HyphenationCompoundWordTokenFilter. 

Lucene issue: https://github.com/apache/lucene/issues/9231
Lucene PR: https://github.com/apache/lucene/pull/12437

### CODE REVIEW

File: modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CompoundAnalysisTests.java [hunk 1]
```diff
@@ -63,6 +67,44 @@ public void testDictionaryDecompounder() throws Exception {
...
+            MatcherAssert.assertThat(
+                terms,
+                hasItems("kaffeemaschine", "kaffee", "fee", "maschine", "fussballpumpe", "fussball", "ballpumpe", "pumpe")
+            );
+        }
+        assertWarnings("Setting [version] on analysis component [custom7] has no effect and is deprecated");
+    }
+
+    /**
+     * Given a word list of: ["kaffee", "fee", "maschine"]
```
Comment (john-wagster) [inline modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CompoundAnalysisTests.java:83]: is that the word list being used or is it this: [fuss, fussball, ballpumpe, ball, pumpe, kaffee, fee, maschine].  I was thrown off by the comment but had trouble tracking that through in my head.  Same thing on the comment on the subsequent test.  The test result makes sense to me and looks good.

Comment (peter-strsr) [reply modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CompoundAnalysisTests.java:83]: Technically, the wordlist contains `["fuss", "fussball", "ballpumpe", "ball", "pumpe", "kaffee", "fee", "maschine"]`, as defined in `test1.json:43`. The comment should highlight, that this parameter should solve this specific problem of preventing the match of "fee" (fairy) within "kaffee" (coffee). 

I left in the same wordlist for all tests and input text to ensure that they are not any unintended side effect. 

If it's clearer I could isolate the tests and only include the _**Kaffeemaschine**_ related words in this test and only the **_Fussballpumpe_** in the other one.

Comment (john-wagster) [reply modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CompoundAnalysisTests.java:83]: gotcha I'm tracking now; this comment was a "for example".   So I'll just nit (change it if you want).  I'd just include before the comment something like "for example given a word list of: " ... that way it's clear that the test is validating more than just that word list.

**Review by john-wagster — APPROVED** [review]
Other than a minor comment this LGTM.

**Comment by john-wagster** [comment]
@elasticsearchmachine please test this

**Comment by john-wagster** [comment]
@elasticmachine test this please

**Comment by john-wagster** [comment]
@elasticmachine update branch

**Pull request merged**